src/ssh.ts

```diff
@@ -0,0 +1,43 @@
+import {info} from '@actions/core'
+import {mkdirP} from '@actions/io'
+import {appendFileSync} from 'fs'
+import {ActionInterface} from './constants'
+import {execute} from './execute'
+import {suppressSensitiveInformation} from './util'
+
+export async function configureSSH(action: ActionInterface): Promise<void> {
+  try {
+    const sshDirectory = `${process.env['HOME']}/.ssh`
+    const sshKnownHostsDirectory = `${sshDirectory}/known_hosts`
+
+    // SSH fingerprints provided by GitHub: https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/githubs-ssh-key-fingerprints
+    const sshGitHubKnownHostRsa =
+      '\ngithub.com ssh-dss AAAAB3NzaC1kc3MAAACBANGFW2P9xlGU3zWrymJgI/lKo//ZW2WfVtmbsUZJ5uyKArtlQOT2+WRhcg4979aFxgKdcsqAYW3/LS1T2km3jYW/vr4Uzn+dXWODVk5VlUiZ1HFOHf6s6ITcZvjvdbp6ZbpM+DuJT7Bw+h5Fx8Qt8I16oCZYmAPJRtu46o9C2zk1AAAAFQC4gdFGcSbp5Gr0Wd5Ay/jtcldMewAAAIATTgn4sY4Nem/FQE+XJlyUQptPWMem5fwOcWtSXiTKaaN0lkk2p2snz+EJvAGXGq9dTSWHyLJSM2W6ZdQDqWJ1k+cL8CARAqL+UMwF84CR0m3hj+wtVGD/J4G5kW2DBAf4/bqzP4469lT+dF2FRQ2L9JKXrCWcnhMtJUvua8dvnwAAAIB6C4nQfAA7x8oLta6tT+oCk2WQcydNsyugE8vLrHlogoWEicla6cWPk7oXSspbzUcfkjN3Qa6e74PhRkc7JdSdAlFzU3m7LMkXo1MHgkqNX8glxWNVqBSc0YRdbFdTkL0C6gtpklilhvuHQCdbgB3LBAikcRkDp+FCVkUgPC/7Rw==\n'
+    const sshGitHubKnownHostDss =
+      '\ngithub.com ssh-dss AAAAB3NzaC1kc3MAAACBANGFW2P9xlGU3zWrymJgI/lKo//ZW2WfVtmbsUZJ5uyKArtlQOT2+WRhcg4979aFxgKdcsqAYW3/LS1T2km3jYW/vr4Uzn+dXWODVk5VlUiZ1HFOHf6s6ITcZvjvdbp6ZbpM+DuJT7Bw+h5Fx8Qt8I16oCZYmAPJRtu46o9C2zk1AAAAFQC4gdFGcSbp5Gr0Wd5Ay/jtcldMewAAAIATTgn4sY4Nem/FQE+XJlyUQptPWMem5fwOcWtSXiTKaaN0lkk2p2snz+EJvAGXGq9dTSWHyLJSM2W6ZdQDqWJ1k+cL8CARAqL+UMwF84CR0m3hj+wtVGD/J4G5kW2DBAf4/bqzP4469lT+dF2FRQ2L9JKXrCWcnhMtJUvua8dvnwAAAIB6C4nQfAA7x8oLta6tT+oCk2WQcydNsyugE8vLrHlogoWEicla6cWPk7oXSspbzUcfkjN3Qa6e74PhRkc7JdSdAlFzU3m7LMkXo1MHgkqNX8glxWNVqBSc0YRdbFdTkL0C6gtpklilhvuHQCdbgB3LBAikcRkDp+FCVkUgPC/7Rw==\n'
+
+    info(`Configuring SSH deploy key… 🔑`)
+
+    await mkdirP(sshDirectory)
+
+    appendFileSync(sshKnownHostsDirectory, sshGitHubKnownHostRsa)
+    appendFileSync(sshKnownHostsDirectory, sshGitHubKnownHostDss)
```

Comment:
Would this be a problem for folks using the js module api? No idea if that and ssh and messing with the user's knownhosts file would be bad?

Reply:
It could be yeah. I wonder what a good alternative would be?

Perhaps we always sshToken to be a string or `true`, and make the boolean skip the token setup and just allow it to context switch to using the ssh endpoints like before? I'd prefer to not have `sshKey` and `ssh` to keep things simpler.

Reply:
Sounds good to me. That would also make the push action re-use ssh auth setups from the checkout action, I guess?

That was another scenario I had in mind.